tests/robustness/pathlock/path_lock_test.go

```diff
@@ -132,9 +135,18 @@ func TestPathLockBasic(t *testing.T) {
 			lock2.Unlock()
 		}()
 
-		time.Sleep(10 * time.Millisecond)
+		// Wait until the internal atomic counter increments.
+		// That will only happen once the Lock call to path2 executes
+		// and blocks on the prior Lock to path1.
+		for {
+			if atomic.LoadUint64(&busyCounter) > currBusyCounter {
+				break
+			}
+
+			time.Sleep(1 * time.Millisecond)
+		}
 
-		if trigger == true {
+		if trigger {
 			t.Fatalf("Lock unsuccessful")
 		}
```

Comment:
This check should be removed  - it is racy and not necessary because you've already confirmed that the goroutine is blocked on the path lock.

Reply:
removed ✔️